### PR TITLE
HTTP Beacon API: `/eth/v1/validator/duties`

### PR DESCRIPTION
**Pull request by rkapka**
**What type of PR is this?**

Feature

**What does this PR do? Why is it needed?**

HTTP version of:
- https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/getAttesterDuties
- https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/getProposerDuties
- https://ethereum.github.io/beacon-APIs/?urls.primaryName=dev#/Validator/getSyncCommitteeDuties

Requesting attester duties of 10 validators for the current epoch takes less than 0.1s
Requesting proposer duties for the current epoch takes less than 0.1s
Requesting sync committee duties of 10 validators for the current epoch takes around 1ms

**Which issues(s) does this PR fix?**

Part of #12635 

**Other notes for review**

### CODE REVIEW

File: beacon-chain/rpc/eth/validator/handlers.go
Comment (james-prysm) [inline beacon-chain/rpc/eth/validator/handlers.go:853]: is this required? I thought the original allowed for empty requests but maybe I missed it.

Comment (rkapka) [reply beacon-chain/rpc/eth/validator/handlers.go:853]: Body is required in the spec. The only other option would be to fetch duties for all validators when it's empty, but this would be rather useless (and very expensive).

Comment (rkapka) [reply beacon-chain/rpc/eth/validator/handlers.go:853]: If the original allowed it, then it most likely didn't return any data. It's better to validate like we do in other endpoints.

File: beacon-chain/rpc/eth/validator/handlers.go
Comment (james-prysm) [inline beacon-chain/rpc/eth/validator/handlers.go:679]: some of these endpoints use this `nextEpoch := currentEpoch + 1`, some don't it would be nice to be more consistent on this.

**Review by james-prysm — COMMENTED** [review]
Mostly ok with this, I didn't find much around it but seems like we might have some old things that could clean up, i can approve if you don't think we need it.

**Comment by terencechain** [comment]
I looked at this today. It looks good to me but I'll let others approve it

**Review by james-prysm — DISMISSED** [review]

**Review by james-prysm — APPROVED** [review]

**Pull request merged**
